packages/grafana-ui/src/components/ColorPicker/ColorPicker.tsx

```diff
@@ -38,6 +38,20 @@ export const colorPickerFactory = <T extends ColorPickerProps>(
       return changeHandler(color);
     };
 
+    stopPropagation = (event: React.KeyboardEvent<HTMLDivElement>, hidePopper: () => void) => {
+      if (event.key === 'Tab' || event.altKey || event.ctrlKey || event.metaKey) {
+        return;
+      }
+
+      event.stopPropagation();
+
+      if (event.key === 'Escape') {
+        hidePopper();
+      }
+
+      return;
+    };
+
```

Comment:
I think I would wait with this and have this PR only solve the focus trap. In later PRs where you need to adress the keyboard navigation you can revisit this.